### DISCUSSION TITLE
about uvicorn workers

### DISCUSSION THREAD

**Discussion by tangent2018** [fastapi/fastapi#6084] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
uvicorn main:app --host 0.0.0.0 --workers 4
```

### Description

Q1:
How can I get the parameters of command configs? In this case, I want workers.
for example: 
>>> from fastapi import xxx
>>> xxx.config.workers
4

Q2:
Can I make a request to a given worker? I want assign a path to an appointed worker.
for example: when request "/some_function", only worker 1 will process it.

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.8

**Comment by iudeen** [comment, marked as answer]
For Q1:
```python
import asyncio

import fastapi
import uvicorn

app = fastapi.FastAPI()


async def main():
    config = uvicorn.Config("main:app", port=5000, log_level="info", workers=4)
    print(config.workers)
    server = uvicorn.Server(config)
    await server.serve()


if __name__ == "__main__":
    asyncio.run(main())
```

For Q2: No, I don't think you can assign a path to a worker. 

What you can rather do is, have it deployed in a different root_path and use a load balancer (like NGINX) to assign the load.

**Comment by tangent2018** [comment]
ok, thank u.

**Reply by BackMountainDevil** [reply]
please mark as answered

**Comment by BackMountainDevil** [comment]
for Q2. you can define the number of worker in config.ini.

```python
uvicorn.Config("main:app", port=5000, log_level="info", workers=readConfig())

app.post(/worker/get):
def getWorker():
    return readConfig()
```
